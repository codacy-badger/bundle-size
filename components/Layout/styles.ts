import css from 'styled-jsx/css';

export const styles = css.global`
  body,
  html {
    background: white;
    max-width: 100vw;
    overflow-x: hidden;
    color: #212121;
  }
  body {
    margin: 0;
  }
  .Layout {
    max-width: 100%;
  }
`;
