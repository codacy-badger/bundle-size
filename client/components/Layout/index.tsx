import { styles } from './styles';
import { ReactNode } from 'react';
import Head from 'next/head';

interface IProps {
  children: ReactNode;
  title?: string;
}

export default function ({ children, title }: IProps) {
  return (
    <>
      {title && (
        <Head>
          <title>{title}</title>
          <meta property="og:title" content={title} key="title" />
        </Head>
      )}

      <div className="Layout__Container">
        <div className="Layout_Wrapper">
          <div className="Layout">{children}</div>
        </div>
      </div>
      <style jsx>{styles}</style>
    </>
  );
}
