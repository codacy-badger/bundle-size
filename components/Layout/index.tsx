import { styles } from './styles';
import { ReactNode } from 'react';

interface IProps {
  children: ReactNode;
}

export default function ({ children }: IProps) {
  return (
    <>
      <div className="Layout">{children}</div>
      <style jsx>{styles}</style>
    </>
  );
}
