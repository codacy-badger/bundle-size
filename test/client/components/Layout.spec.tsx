import React from 'react';
import { render } from '../../testUtils';
import Layout from '../../../client/components/Layout';

describe('Layout component', () => {
  it('matches snapshot', () => {
    const props = { children: '' };
    const { asFragment } = render(<Layout {...props} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
