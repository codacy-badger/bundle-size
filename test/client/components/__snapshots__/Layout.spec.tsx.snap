// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-13839950 Layout__Container"
  >
    <div
      class="jsx-13839950 Layout_Wrapper"
    >
      <div
        class="jsx-13839950 Layout"
      />
    </div>
  </div>
</DocumentFragment>
`;
